kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+    using SparseMatrixType = typename BaseType::SparseMatrixType;
+
+    /// Type definition for pointer to sparse matrix
+    using SparseMatrixPointerType = typename BaseType::SparseMatrixPointerType;
+
+    /// Type definition for vector
+    using VectorType = typename BaseType::VectorType;
+
+    /// Type definition for pointer to vector
+    using VectorPointerType = typename BaseType::VectorPointerType;
+
+    /// Type definition for dense matrix
+    using DenseMatrixType = typename BaseType::DenseMatrixType;
+
+    /// Type definition for dense vector
+    using DenseVectorType = typename BaseType::DenseVectorType;
+
+    /// Type definition for size
+    using SizeType = std::size_t;
+
+    /// Type definition for index
+    using IndexType = std::size_t;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief This is the default constructor
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(Parameters ThisParameters = Parameters(R"({})"))
+        : mParameters(ThisParameters)
+    {
+        // Set the default parameters
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Define the solvers
+        const SizeType number_of_solver_parameters = mParameters["solvers"].size();
+        KRATOS_ERROR_IF(number_of_solver_parameters == 0) << "No solvers defined in the parameters" << std::endl;
+        mSolvers.reserve(number_of_solver_parameters);
+        for (IndexType i = 0; i < number_of_solver_parameters; ++i) {
+            mSolvers.push_back(ConstructLinearSolverFromSettings(mParameters["solvers"][i]));
+        }
+
+        // Set some member variables from the parameters
+        CommonSettingsFromParameters();
+    }
+
+    /**
+     * @brief Constructor with two solvers
+     * @param pSolver1 The first LinearSolverPointer to set.
+     * @param pSolver2 The second LinearSolverPointer to set.
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(
+        LinearSolverPointer pSolver1,
+        LinearSolverPointer pSolver2,
+        Parameters ThisParameters = Parameters(R"({})")
+        ) : mSolvers({pSolver1, pSolver2}),
+            mParameters(ThisParameters)
+    {
+        // Set the default parameters
+        mParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+        // Fill the parameters with the solvers
+        FillParametersFromSolver(pSolver1);
+        FillParametersFromSolver(pSolver2);
+
+        // Set some member variables from the parameters
+        CommonSettingsFromParameters();
+    }
+
+    /**
+     * @brief Constructor with a list of solvers
+     * @param rSolvers A vector of LinearSolverPointer to set.
+     * @param ThisParameters The configuration parameters
+     */
+    FallbackLinearSolver(
+        const std::vector<LinearSolverPointer>& rSolvers,
+        Parameters ThisParameters = Parameters(R"({})")
+        ) : mSolvers(rSolvers),
```

Comment:
is there some use case where passing a list of solvers **and** the parameters is necessary? In my opinion, it only creates more possibility for confusion and errors (e.g.: the parameters also define a list of solvers that are different than the ones contained in the vector).